specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json

```diff
@@ -1359,9 +1361,249 @@
     "CacheSecuritySettings": {
       "description": "Cache security settings.",
       "properties": {
-        "rootSquash": {
+        "accessPolicies": {
+          "description": "NFS access policies defined for this cache.",
+          "items": {
+            "$ref": "#/definitions/NfsAccessPolicy"
+          },
+          "type": "array"
+        }
+      }
+    },
+    "CacheDirectorySettings": {
+      "description": "Cache Directory Services settings.",
+      "type": "object",
+      "properties": {
+        "activeDirectory": {
+          "$ref": "#/definitions/CacheActiveDirectorySettings",
+          "description": "Specifies the settings for joining the HPC Cache to an Active Directory domain."
+        },
+        "usernameDownload": {
+          "$ref": "#/definitions/CacheUsernameDownloadSettings",
+          "description": "Specifies the settings for Extended Groups. Extended Groups allows users to be members of more than 16 groups."
+        }
+      }
+    },
+    "CacheActiveDirectorySettings": {
+      "description": "Active Directory settings used to join a Cache to a Domain.",
+      "properties": {
+        "primaryDnsIpAddress": {
+          "description": "primary DNS IP address of the AD Domain controller.",
+          "type": "string"
+        },
+        "secondaryDnsIpAddress": {
+          "description": "secondary DNS IP address of the AD Domain controller.",
+          "type": "string"
+        },
+        "domainName": {
+          "description": "The Fully Qualified Domain Name of the Active Directory Domain.",
+          "type": "string"
+        },
+        "domainNetBios": {
+          "description": "The Active Directory Domain's NetBios.",
+          "type": "string"
+        },
+        "smbServerName": {
+          "description": "The name (NetBIOS) used for the HPC Cache to join the Active Directory Domain. Length must be not greater than 15 and chars must be in list of [-0-9a-zA-Z_] char class.",
```

Comment:
The name is used to join the domain and for smb shares that would live on the hpc cache in the future.
Would prefer not to change it again once we add the smb api. Perhaps we can add to the description indicating that will be used in the future, but I'm not sure what the PMs think about that.

Reply:
Do we need a change here? Field name and/or description?

We're using this field now for non-SMB use, correct? If so, there is some non-future use here. Deferring to PM about the future use clause.

Just trying to make this clear, but not quite sure how.

@scjescho @nydickmsft @waffe7 @brpanask (Sending link to Erin, too.)

Reply:
@ekpgh

Reply:
All I could come up with is the following minor tweak for `smbServerName`'s description so that it follows the description for `domainNetBios` (The Active Directory domain's NetBIOS name.):

`The HPC Cache's NetBIOS name, which used to join the Active Directory domain. Length must not be greater than 15 and chars must be from the [-0-9a-zA-Z] char class.`

I doubt it's desirable, but something like this can be added between the two sentences to better clarify the link between the field's name and the description: `This will be the cache's SMB server name when SMB support is added in the future.`

This would pair up the NetBIOS fields as such:
> **domainNetBios**: The Active Directory domain's NetBIOS name.
> **smbServerName**: The HPC Cache's NetBIOS name, which used to join the Active Directory domain. Length must not be greater than 15 and chars must be from the [-0-9a-zA-Z] char class.


...

In that vein, one last attempt suggesting altering the field name to follow the (clearer) pattern of domainNetBios: `smbServerName --> cacheNetBios`
> **domainNetBios**: The Active Directory domain's NetBIOS name.
> **cacheNetBios**: The HPC Cache's NetBIOS name, which used to join the Active Directory domain. Length must not be greater than 15 and chars must be from the [-0-9a-zA-Z] char class.

Then when adding SMB support, just tweak the description:
> **domainNetBios**: The Active Directory domain's NetBIOS name.
> **cacheNetBios**: The HPC Cache's NetBIOS name, which used to join the Active Directory domain. _**This is the cache's SMB server name.**_ Length must not be greater than 15 and chars must be from the [-0-9a-zA-Z] char class.

My two cents above. Final call, @scjescho @Juarezhm @ekpgh @waffe7 @nydickmsft @mannywarrior ?

Reply:
Resolved internally.